qiskit/optimization/algorithms/minimum_eigen_optimizer.py

```diff
@@ -11,43 +11,74 @@
 
 """A wrapper for minimum eigen solvers from Aqua to be used within the optimization module."""
-from typing import Optional, Any, Union, Tuple, List, cast
+from collections import namedtuple
+from typing import Optional, Any, Union, List, cast
 
 import numpy as np
+
 from qiskit.aqua.algorithms import MinimumEigensolver, MinimumEigensolverResult
 from qiskit.aqua.operators import StateFn, DictStateFn
-
 from .optimization_algorithm import (OptimizationResultStatus, OptimizationAlgorithm,
                                      OptimizationResult)
 from .. import QiskitOptimizationError
 from ..converters.quadratic_program_to_qubo import QuadraticProgramToQubo, QuadraticProgramConverter
 from ..problems.quadratic_program import QuadraticProgram, Variable
 
+RawSolutionSample = namedtuple('RawSolutionSample', 'bitstring fval probability')
+RawSolutionSample.__doc__ = """\
+Raw solution sample
+
+    bitstring: the basis sate
+    fval: the QUBO value
+    probability: the probability
+"""
+
+SolutionSample = namedtuple('SolutionSample', 'x fval probability status')
+SolutionSample.__doc__ = """\
+Solution sample interpreted for the original problem
+
+    x: the values of variables
+    fval: the objective function value of the original problem
+    probability: the probability
+    status: the status
+"""
+
 
 class MinimumEigenOptimizationResult(OptimizationResult):
     """ Minimum Eigen Optimizer Result."""
 
     def __init__(self, x: Union[List[float], np.ndarray], fval: float, variables: List[Variable],
-                 status: OptimizationResultStatus, samples: List[Tuple[str, float, float]],
+                 status: OptimizationResultStatus,
+                 samples: List[SolutionSample],
+                 raw_samples: List[RawSolutionSample],
                  min_eigen_solver_result: Optional[MinimumEigensolverResult] = None) -> None:
         """
         Args:
             x: the optimal value found by ``MinimumEigensolver``.
             fval: the optimal function value.
             variables: the list of variables of the optimization problem.
             status: the termination status of the optimization algorithm.
-            samples: the basis state as bitstring, the QUBO value, and the probability of sampling.
+            samples: the x value, the objective function value of the original problem,
+                the probability, and the status of sampling.
+            raw_samples: the basis state as bitstring, the QUBO value, and the probability of
+                sampling.
             min_eigen_solver_result: the result obtained from the underlying algorithm.
         """
         super().__init__(x, fval, variables, status, None)
         self._samples = samples
+        self._raw_samples = raw_samples
         self._min_eigen_solver_result = min_eigen_solver_result
 
     @property
-    def samples(self) -> List[Tuple[str, float, float]]:
+    def samples(self) -> List[SolutionSample]:
```

Comment:
1. Can we instead of `samples()` use `sampled_solutions()`?
2. We need this property in the base class, in `OptimizationResult`, and a default implementation returns a list of one value (suitable for classic optimizers and others with just one solution).

Reply:
Yes, I'll move it to `OptimizationResult` and rename it as you suggested.